sda/service.go

```diff
@@ -72,6 +72,7 @@ type serviceFactory struct {
 	translations map[string]ServiceID
 	// Inverse mapping of ServiceId => string
 	inverseTr map[ServiceID]string
+	mutex     sync.Mutex
```

Comment:
Can you just remind me again why the deadlock / data race is happening ? Because normally, you register `Service` in every `init()` methods so I don't see how there could be a data race.  Is it because of the tests ?